docs/how-to-guides/develop-a-flex-flow/function-based-flow.md

```diff
@@ -0,0 +1,98 @@
+# Function based flow
+
+User can directly use a function(see [supported types](./index.md#supported-types) for typing support) as flex flow's entry.
+**Note**: the annotation is not required, user can use `flow save` to provide signatures or write `flow.flex.yaml`.
+
+## Authoring
+
+```python
+class Reply(TypedDict):
+    output: str
+
+@trace
+def my_flow(text: str) -> Reply: 
+    pass
+```
+
+## YAML support
+
+Similar as DAG flow. YAML file is identifier for flex flow.
+Flex flow will use `flow.flex.yaml` as it's identifier.
+User can write the YAML file manually or save a function/callable entry to YAML file.
+A complete flex flow YAML may look like this:
+
+```yaml
+$schema: https://azuremlschemas.azureedge.net/promptflow/latest/Flow.schema.json
+entry: path.to.module:function_name
+inputs:
```

Comment:
don't mention inputs outputs for now. As we are defining sample & signature field for flex-flow